packages/flutter/lib/src/cupertino/page_scaffold.dart

```diff
@@ -59,22 +69,28 @@ class CupertinoPageScaffold extends StatelessWidget {
 
       // TODO(xster): Use real size after partial layout instead of preferred size.
       // https://github.com/flutter/flutter/issues/12912
-      final double topPadding = navigationBar.preferredSize.height
-          + existingMediaQuery.padding.top;
+      final double topPadding =
+          navigationBar.preferredSize.height + existingMediaQuery.padding.top;
+
+      // Propagate bottom padding and include viewInsets if appropriate
+      final double bottomPadding = resizeToAvoidBottomPadding
+              ? existingMediaQuery.viewInsets.bottom
+              : 0.0;
 
       // If navigation bar is opaquely obstructing, directly shift the main content
       // down. If translucent, let main content draw behind navigation bar but hint the
       // obstructed area.
       if (navigationBar.fullObstruction) {
         paddedContent = new Padding(
-          padding: new EdgeInsets.only(top: topPadding),
+          padding: new EdgeInsets.only(top: topPadding, bottom: bottomPadding),
           child: child,
         );
       } else {
         paddedContent = new MediaQuery(
           data: existingMediaQuery.copyWith(
             padding: existingMediaQuery.padding.copyWith(
               top: topPadding,
+              bottom: existingMediaQuery.padding.bottom + bottomPadding,
```

Comment:
Reading this a bit more carefully, I see what's happening in the tests. Instead of converting MediaQuery insets into MediaQuery paddings, we want to directly apply the insets to the contents. 

In other words, leave the MediaQuery part as is but wrap that whole thing with a Padding with a bottom padding from the inset. This matches the Material scaffold's behavior more and you can remove the SafeAreas in the test.